src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -735,7 +735,12 @@ public void execute() throws GitException, InterruptedException {
 
                 args.add(fastForwardMode);
                 args.add(rev.name());
-                launchCommand(args);
+
+                String repoUrl = getRemoteUrl(getDefaultRemote());
```

Comment:
Merge is allowed even if a remote URL is not defined.  If there is no remote URL, there is no need to use credentials in the merge.

I think the credentials should only be used in the merge if there is a remote URL, otherwise launch the command without credentials.